liquibase-standard/src/main/java/liquibase/util/StandardSqlParser.java

```diff
@@ -48,17 +53,52 @@ public StringClauses parse(String sqlBlock, boolean preserveWhitespace, boolean
                 }
                 token = t.getNextToken();
             }
-
         } catch (Throwable e) {
             if (changeSet != null) {
-                throw new UnexpectedLiquibaseException(changeSet.toString(), e);
+                Change change = Scope.getCurrentScope().get(ChangeSet.CHANGE_KEY, Change.class);
+                String message = enhanceExceptionMessage(changeSet, change, e);
+                throw new UnexpectedLiquibaseException(message, e);
             } else {
                 throw new UnexpectedLiquibaseException(e);
             }
         }
         return clauses;
     }
 
+    /**
+     *
+     * If this is a RawSQLChange, then add information about the real position of the SQL in the
+     * formatted SQL changelog
+     *
+     * @param  changeSet     The current change set
+     * @param  change        The change that generated the SQL
+     * @param  e             The thrown exception
+     * @return String        The message to display
+     *
+     */
+    private static String enhanceExceptionMessage(ChangeSet changeSet, Change change, Throwable e) {
+        if (! (change instanceof RawSQLChange)) {
+            return changeSet.toString();
+        }
+        String exceptionMessage = e.getMessage();
+        Pattern p = Pattern.compile("(?i).* line ([\\d]+).*");
```

Comment:
I thought you weren't going to try to parse out the line number from the exception message? Or was it simple enough that you figured we might as well try this first?

Reply:
Yes, I thought it was straight-forward enough to try to match on it.  Worse case is that we can still display the range of lines from the changelog.

Reply:
This is a good approach. Postgres will do a better job of identifying the line number than we can likely do.